pos_report_order_payment/report/pos_order_payment_report.py

```diff
@@ -148,9 +148,10 @@ def _payment_query(self):
                 pol.id AS pos_line_id,
                 sl.id AS statement_line_id,
                 st.journal_id AS journal_id,
-                (
-                    sl.amount / po.amount_total * pol.price_subtotal_incl
-                ) AS amount
+                CASE po.amount_total
+                    WHEN 0 THEN 0
+                    ELSE sl.amount / po.amount_total * pol.price_subtotal_incl
```

Comment:
And if price_subtotal_incl is 0. Can occures with a discount of 100%. Dont'you think ?

Otherwise. Lgtm. Thanks a lot !

Reply:
That's actually evaluated as `(sl.amount / po.amount_total) * pol.price_subtotal_incl`, so it's not an issue

You can try it with:
`psql -c "select 1 / 1 * 0"`

Reply:
Oh ! You're right.
Thanks.